### DISCUSSION TITLE
Feature: Resolve isoa2 country code

### DISCUSSION THREAD

**Discussion by steffenrh** [jannikmi/timezonefinder#177] [Ideas]
Hello Jannik,

do you think it's possible to resolve the country and return the isoa2 code  (or some UNDEF/NONE in case of unclaimed sea area) as result set together with the time zone? Before utilizing your timezonefinder (THANK YOU!) I used a command line tool for our resolving issues, and that one (among other values) returned the timezone and the country code. Having the country code would be a great help, as we have the country code in our input data and we could compare the two country codes and warn the data owner in case the input data country code differs from the timezone country code.

Would be great if timezonefinder could provide this additional data!

cu, Steffen

**Comment by jannikmi** [comment]
That would be an easy addition if the mapping from timezone to iso country code was available. Do you have an idea how to obtain/compute this mapping? (It has to be accurate)

**Comment by steffenrh** [comment]
I actually don't know how exactly this is working. But I've just downloaded this project [https://github.com/BertoldVdb/ZoneDetect](https://github.com/BertoldVdb/ZoneDetect) and compiled the demo.c for me. Output looks like this:

```
./demo ../database/timezone21.bin 48.7208805 9.122417
In zone:
  meta: 6424
  TimezoneIdPrefix: Europe/
  TimezoneId: Berlin
  CountryAlpha2: DE
  CountryName: Germany
Safezone: 0.922633
```

Unfortunately I'm lousy in C **and** in math, so I don't really understand what's happening in `zonedetect.c`. :-)  But obviously the polygon database used by this project contains the necessary data to resolve country information.

**Reply by jannikmi** [reply]
Interesting. Can you please double check if CountryAlpha2 is the ISO code you are looking for?

**Comment by steffenrh** [comment]
Definitely. I used this solution for almost 3 years in production, now I replaced this with a Python rewrite importing your timezonefinder. But the lack of the CountryAlpha2 is a little bit of a regression for my "double check features" in the code.

**Reply by jannikmi** [reply]
Cool. And could you please try to track down how and where the library stores this mapping? We might then we able to simply reuse it in timezonefinder.

**Comment by jannikmi** [comment]
Unfortunately this has become impossible with the use of the reduced set of timezones starting with version 8.0.0
